zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -50,12 +56,53 @@ where
         .or_else(|| params.activation_height(NetworkUpgrade::Sapling))
         .ok_or(Error::SaplingNotActive)?;
 
+    #[cfg(feature = "transparent-inputs")]
+    let utxos_spent = {
+        let mut addrs = HashSet::new();
+        for account_id in ufvks.keys() {
+            for addr in data.get_transparent_receivers(*account_id)? {
+                addrs.insert(addr);
+            }
+        }
+
+        tx.transparent_bundle()
+            .iter()
+            .flat_map(|b| b.vin.iter())
+            .filter_map(|txin| {
+                // If the txin belongs to one of our addresses
+                if txin
+                    .script_sig
+                    .address()
```

Comment:
This doesn't work, sorry 😅 `Script::address` is written to match on the contents of a `scriptPubKey` (which has a well-defined format for P2PKH and P2SH which directly contains the address), not a `scriptSig` (which only makes sense in the context of a `scriptPubKey`). We could in theory extract an address from a `scriptSig` (hash the pubkey in the case of a P2PKH redeem script, and hash the included script in the case of a P2SH redeem script), but we technically do not know whether a `scriptSig` is for a P2PKH or P2SH output without actually looking up that output (it could in fact be for some other output script that is not P2PKH or P2SH but happens to have a similar-looking `scriptSig`).

There are two things we should do:
- For this PR, we should extract the full list of outpoints from the transaction, and then call the data access API to determine whether any of the UTXOs appear in our data store (from which we can set `spends_our_utxos` and then either filter the list there, or just pass the full list over for the data access API to update only the ones that exist).
- We should document `Script::address` to indicate that it is for use specifically on `scriptPubKey`. We could maybe also just move it onto `TxOut` instead (making the existing method crate-private and exposing it publicly via a wrapper method on the output type).